pkg/kubelet/kubelet.go

```diff
@@ -1629,6 +1665,30 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 		return errOuter
 	}
 
+	// Create Cgroups for the pod and apply resource parameters to them if cgroup-per-qos flag is enabled.
+	pcm := kl.containerManager.NewPodContainerManager()
+	// If pod has already been terminated then we need not create or update the pod's cgroup
+	if !kl.podIsTerminated(pod) {
```

Comment:
Oh I understand your concern. This would unnecessarily create the pod cgroup's. Would make the change!